torch/csrc/jit/python/init.cpp

```diff
@@ -1058,6 +1136,95 @@ void initJITBindings(PyObject* module) {
         }
       });
 
+  py::class_<c10::SymbolicIntNode, std::shared_ptr<c10::SymbolicIntNode>>(m, "SymbolicIntNode")
+    .def_static("new_symint", [](py::object obj) -> std::shared_ptr<c10::SymbolicIntNode> {
+      return std::make_shared<PythonSymbolicIntNode>(obj);
+    })
+    .def_static("isinstance", [](py::object obj, bool convert) -> bool {
+      return pybind11::detail::type_caster<std::shared_ptr<c10::SymbolicIntNode>>().load(obj, convert);
+      //return false;
+    })
+    // FOREACH MACRO?
+    .def("__add__", [](std::shared_ptr<c10::SymbolicIntNode> a, py::object b) -> std::shared_ptr<c10::SymbolicIntNode> {
+      if (torch::is_symint_node(b)) {
+        return std::shared_ptr<c10::SymbolicIntNode>(a->add(b.cast<c10::SymbolicIntNode*>()));
+      } else {
+        return std::shared_ptr<c10::SymbolicIntNode> (a->add(a->wrap(b.cast<int64_t>())));
```

Comment:
This here feels like a helper function would help a bit here. But it's also not entirely clear you want to wrap integers into symbolic int nodes (that denote plain integers); it seems like it would be more user friendly if these showed up at dispatch site as plain integers. It might be a bit easier here to make the add method accept an IValue instead of a SymbolicIntNode, so you can pass in either an int or symbolic int without needing to unconditionally accept a SymbolicIntNode.

Reply:
it looks way nicer rn.
I don't want to introduce a dependency on `IValue` into `SymbolicIntNode` :( . It seems more complex architecturally and possibly less user friendly since both LTC and AOTAutograd will need to parse `IValue`s explicitly. Both LTC and AOTAutograd already wrap ints into `sympy.Integer` or `prim::Constant`.